clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinatorTest.java

```diff
@@ -3114,6 +3137,70 @@ public void testFetchCommittedOffsets() {
         assertEquals(new OffsetAndMetadata(offset, leaderEpoch, metadata), fetchedOffsets.get(t1p));
     }
 
+    @Test
+    public void testPopulatingOffsetCacheForAssignedPartition() {
+        client.prepareResponse(groupCoordinatorResponse(node, Errors.NONE));
+        coordinator.ensureCoordinatorReady(time.timer(Long.MAX_VALUE));
+
+        Map<TopicPartition, OffsetAndMetadata> committedOffsetsCache = coordinator.committedOffsetsCache();
+        // committedOffsetsCache should be empty
+        assertEquals(committedOffsetsCache.size(), 0);
+
+        long offset = 500L;
+        String metadata = "blahblah";
+        Optional<Integer> leaderEpoch = Optional.of(15);
+        OffsetFetchResponse.PartitionData data = new OffsetFetchResponse.PartitionData(offset, leaderEpoch,
+                metadata, Errors.NONE);
+
+        client.prepareResponse(offsetFetchResponse(Errors.NONE, singletonMap(t1p, data)));
+        subscriptions.assignFromUser(singleton(t1p));
+        Map<TopicPartition, OffsetAndMetadata> fetchedOffsets = coordinator.fetchCommittedOffsets(singleton(t1p),
+                time.timer(Long.MAX_VALUE));
+
+        assertNotNull(fetchedOffsets);
+        OffsetAndMetadata expected = new OffsetAndMetadata(offset, leaderEpoch, metadata);
+        assertEquals(expected, fetchedOffsets.get(t1p));
+
+        // check committedOffsetsCache is populated
+        assertEquals(committedOffsetsCache.size(), 1);
+        assertEquals(expected, committedOffsetsCache.get(t1p));
+    }
+
+    @Test
+    public void testReturningCachedOffsetForAssignedPartition() {
+        client.prepareResponse(groupCoordinatorResponse(node, Errors.NONE));
+        coordinator.ensureCoordinatorReady(time.timer(Long.MAX_VALUE));
+
+        Map<TopicPartition, OffsetAndMetadata> committedOffsetsCache = coordinator.committedOffsetsCache();
+
+        long offset = 500L;
+        String metadata = "blahblah";
+        Optional<Integer> leaderEpoch = Optional.of(15);
+        OffsetAndMetadata offsetAndMetadata = new OffsetAndMetadata(offset, leaderEpoch, metadata);
+        committedOffsetsCache.put(t1p, offsetAndMetadata);
```

Comment:
I'm thinking we can merge this test with the above `testPopulatingOffsetCacheForAssignedPartition` test, so that we don't have to initialize all the data again. Something like this:
```
// check committedOffsetsCache is populated
        assertEquals(committedOffsetsCache.size(), 1);
        assertEquals(expected, committedOffsetsCache.get(t1p));

       
       client.prepareResponse(offsetFetchResponse(Errors.NONE, singletonMap(t2p, data)));
       // fetch again with t1p + t2p, but will send fetch for t2p since t1p is in cache
        Map<TopicPartition, OffsetAndMetadata> fetchedOffsets = coordinator.fetchCommittedOffsets(new HashSet<>(Arrays.asList(t1p, t2p)),
                time.timer(Long.MAX_VALUE));

        assertNotNull(fetchedOffsets);
        // return 2 results
        assertEquals(fetchedOffsets.size(), 2);
        // the cache size is still 1
        assertEquals(committedOffsetsCache.size(), 1);
 .....

```
WDYT?